src/dune/action_ast.ml

```diff
@@ -180,6 +182,7 @@ struct
           ; ( "cmp"
             , let+ diff = Diff.decode_binary path in
               Diff diff )
+          ; ("no-infer", t >>| fun t -> No_infer t)
```

Comment:
Ah, so looking at the self-contained version of the test made me notice one issue: `(no-infer ...)` shouldn't be accepted with `(lang dune 2.0)` given that this feature was introduced in Dune 2.6. Here is the fix:

```suggestion
          ; ("no-infer", Dune_lang.Syntax.since Stanza.syntax (2, 6) >>> t >>| fun t -> No_infer t)
```